### PR TITLE
DOC: Improve ``poisson`` plot

### PR DESCRIPTION
**Pull request by linus-md**
This PR adds the pmf to the plot and add some example calculations.

### CODE REVIEW

File: numpy/random/_generator.pyx [hunk 1]
```diff
@@ -3305,14 +3305,21 @@ cdef class Generator:
...
 
-        >>> import numpy as np
         >>> rng = np.random.default_rng()
-        >>> s = rng.poisson(5, 10000)
+        >>> lam, size = 50, 10000
+        >>> s = rng.poisson(lam=lam, size=size)
 
-        Display histogram of the sample:
+        The mean and variance of the distribution are both ``lam``
+        >>> s.mean(), s.var()
```
Comment (rkern) [inline numpy/random/_generator.pyx:3313]:
```suggestion

        >>> s.mean(), s.var()
```

File: numpy/random/_generator.pyx [hunk 2]
```diff
@@ -3305,14 +3305,22 @@ cdef class Generator:
...
+        (50.0639, 48.63461679)  # may vary
+
+        The sample with the corresponding pmf 
 
         >>> import matplotlib.pyplot as plt
-        >>> count, bins, _ = plt.hist(s, 14, density=True)
+        >>> from scipy import stats  # doctest: +SKIP
+        >>> count, bins, _ = plt.hist(s, 30, density=True)
+        >>> x = np.arange(0, 100)
+        >>> plt.plot(x, stats.poisson.pmf(x, mu=lam))
```
Comment (emltoja) [inline numpy/random/_generator.pyx:3323]: Shouldn't we use here `stem` or `scatter` plot as this is *pmf* not *pdf*?  Support of the Poisson distribution is $\mathbb{N}$ not  $\mathbb{R}$.

File: numpy/random/_generator.pyx [hunk 3]
```diff
@@ -3305,14 +3305,22 @@ cdef class Generator:
         --------
         Draw samples from the distribution:
 
-        >>> import numpy as np
         >>> rng = np.random.default_rng()
-        >>> s = rng.poisson(5, 10000)
+        >>> lam, size = 50, 10000
```
Comment (emltoja) [inline numpy/random/_generator.pyx:3309]:
```suggestion
        >>> lam, size = 5, 10000
```

Change `lam` to make plot more clear

File: numpy/random/_generator.pyx [hunk 4]
```diff
@@ -3305,14 +3305,22 @@ cdef class Generator:
...
+        >>> s = rng.poisson(lam=lam, size=size)
 
-        Display histogram of the sample:
+        The mean and variance of the distribution are both ``lam``
+        
+        >>> s.mean(), s.var()
+        (50.0639, 48.63461679)  # may vary
```
Comment (emltoja) [inline numpy/random/_generator.pyx:3315]:
```suggestion
        (4.9917 5.1088311)  # may vary
```

File: numpy/random/_generator.pyx [hunk 5]
```diff
@@ -3305,14 +3305,22 @@ cdef class Generator:
...
+
+        The sample with the corresponding pmf 
 
         >>> import matplotlib.pyplot as plt
-        >>> count, bins, _ = plt.hist(s, 14, density=True)
+        >>> from scipy import stats  # doctest: +SKIP
+        >>> count, bins, _ = plt.hist(s, 30, density=True)
```
Comment (emltoja) [inline numpy/random/_generator.pyx:3321]:
```suggestion
        >>> weights = np.ones_like(s)/len(s)
        >>> count, bins, _ = plt.hist(s, 30, weights=weights)
```

This is a **pmf** estimator, not a **pdf** one.

File: numpy/random/_generator.pyx [hunk 6]
```diff
@@ -3305,14 +3305,22 @@ cdef class Generator:
...
+        >>> x = np.arange(0, 100)
```
Comment (emltoja) [inline numpy/random/_generator.pyx:3322]:
```suggestion
        >>> x = np.arange(0, 20)
```

File: numpy/random/_generator.pyx (hunk 2 above)
Comment (emltoja) [inline numpy/random/_generator.pyx:3323]:
```suggestion
        >>> plt.stem(x, stats.poisson.pmf(x, mu=lam), linefmt='C1-')
```

**Review by emltoja — COMMENTED** [review]
looks great to me

**Review by mdhaber — COMMENTED** [review]
Sometimes the plot looks undesirable like 
image

I'm pushing changes that fix this and make the wording more consistent with related docstrings. I removed the `# doctest: +SKIP` as an experiment; we'll see if it's needed.

**Pull request merged**